### PR TITLE
Fixes the Adhomian watch's time/date calculation.

### LINKED ISSUE #10373: Adhomian watch readout issue.
When the Adhomian watch is examined, it gives you the time and date on Adhomai. It seems like the Adhomian day is not displaying properly though. Examining it will give a message along the lines of "You check your adhomian watch, glancing over at the watch face, reading the time to be '10. Today's date is 'th day of Sil'nrry-rhazzimy 3620". It seems the day is lacking, as it never gets numbered.

### PR DESCRIPTION
**Pull request by MarinaGryphon**
Fixes #10373

### CODE REVIEW

File: code/game/objects/items/tajara.dm [hunk 1]
```diff
@@ -146,47 +147,23 @@
 	else
 
 		var/adhomian_year = game_year + 1158
-
-		var/current_month = time2text(world.realtime, "Month")
+		var/current_month = text2num(time2text(world.realtime, "MM"))
 		var/current_day = text2num(time2text(world.realtime, "DD"))
 		var/adhomian_day
-		var/adhomian_month
+		var/adhomian_month = src.months[Ceiling(current_month/3)]
```
Comment (Amunak) [inline code/game/objects/items/tajara.dm:153]: Using modulo might be more appropriate.

Assuming a range of current_month from 1 to 12 this should work:
```suggestion
		var/adhomian_month = src.months[current_month - 1 % 4]
```

File: code/game/objects/items/tajara.dm [hunk 2]
```diff
@@ -135,6 +135,7 @@
 	icon_state = "adhomai_clock"
 	item_state = "adhomai_clock"
 	contained_sprite = TRUE
+	var/static/months = list("Menshe-aysaif","Sil'nryy-aysaif","Menshe-rhazzimy","Sil'nryy-rhazzimy")
```
Comment (Amunak) [inline code/game/objects/items/tajara.dm:138]: This is a tiny nitpick, but I _think_ we usually style stuff with spaces after commas?

```suggestion
	var/static/months = list("Menshe-aysaif", "Sil'nryy-aysaif", "Menshe-rhazzimy", "Sil'nryy-rhazzimy")
```

File: code/game/objects/items/tajara.dm [hunk 3]
```diff
@@ -146,47 +147,23 @@
...
-					adhomian_day = current_day + 15
-				if (current_month == "December")
-					adhomian_day = current_day + 30
-
+			if(2, 5, 8, 11)
+				current_day += 31
+			if(6, 9, 12)
+				current_day += 61
+			if(3)
+				current_day += 59 + isLeap(text2num(time2text(world.realtime, "YYYY")))
```
Comment (Amunak) [inline code/game/objects/items/tajara.dm:160]: As discussed on Discord I'm not really convinced this is explicit enough. It looks like a mistake.

I'd either add a comment like this:
```suggestion
				current_day += 59 + isLeap(text2num(time2text(world.realtime, "YYYY"))) // we can conveniently use the result of `isLeap` to add 1 when we are in a leap year
```

or used a ternary to be more explicit:

```suggestion
				current_day += 59 + ( isLeap(text2num(time2text(world.realtime, "YYYY"))) ? 1 : 0)
```

or I'd separate it into branches:

```suggestion
				if(isLeap(text2num(time2text(world.realtime, "YYYY"))))
					current_day += 60
				else
					current_day += 59
```

Pick your poison 🙂.

**Review by Amunak — APPROVED** [review]

**Review by Chada-1 — APPROVED** [review]

File: code/game/objects/items/tajara.dm (hunk 1 above)
Comment (MarinaGryphon) [reply code/game/objects/items/tajara.dm:153]: Not equivalent. I think the correct one would be `(current_month - 1) % 3 + 1`, I considered doing a couple (micro-?)optimisations but this isn't hot code or anything, so I don't think it matters much.

Comment (Amunak) [reply code/game/objects/items/tajara.dm:153]: Oh I see, I didn't understand the calendar properly then. I thought they have only 4-month year, and instead it's a "regular" year where each month is 3 times as long. Carry on.

**Review by Alberyk — APPROVED** [review]

**Pull request merged**